### ISSUE TITLE
Configuratyion for speedtest-cli

### ISSUE THREAD

**Issue by Scoperius** [XavierBerger/RPi-Monitor#383]
Hi I tried to make a running configuration for speedtest-cli (https://github.com/sivel/speedtest-cli). But without any success. Can you please point me the right way? May proposed conf does not display anything: 

dynamic.101.name=Ping
dynamic.101.source=/home/pi/speedtest-cli/speedtest.py --simple | grep "Ping: "
dynamic.101.regexp=(.*)

web.status.1.content.10.line.2="Internet Ping: <b>" + data.Ping

although /home/pi/speedtest-cli/speedtest.py --simple | grep "Ping: " returns Ping value properly 

Thanks for help,

Petr

**Comment by dirdled** [comment]
Hi Scoperius, I'm not sure if it has to be this way, but I had the best luck when I put just the command for the source, and then did the filtering using the regexp line, so for your config I would try something like

dynamic.101.name=Ping
dynamic.101.source=/home/pi/speedtest-cli/speedtest.py --simple
dynamic.101.regexp=(Ping:.*)
dynamic.101.postprocess=$1

Also there may be a better way to do this, but when I wanted to monitor my internet stats using speedtest-cli and rpimonitor, I built a wrapper for the speedtest command, because it takes so long to run. I think you can also use the interval and timeout values to do it, but I had other reasons why I wanted to use a wrapper. Basically what my wrapper does is checks the age (and existence) of the previous output file, and only runs the speedtest if the file is older than a certain age. Then rpimonitor is able to read that output file as often as it likes without having to worry about
a) Creating a lot of extra bandwidth
and
b) How long the command takes to run.

Like I said there may be better ways to do all of this, but that is what worked for me

**Comment by sgrayban** [comment]
Care to share your wrapper?

**Comment by sgrayban** [comment]
@dirdled Care to share your wrapper?

**Comment by dirdled** [comment]
> 
Hi sgrayban, 

I can share, but I'm not sure my code is suitable for mass consumption. I tend to do Quick and dirty stuff, with the intention of going back and polishing it if I ever run out of projects and have spare time...So far that hasn't happened

**Comment by sgrayban** [comment]
@dirdled

Well I was just wondering how you managed to remove the text out of the speedtest results.

Currently I have a cron running every 10 minutes that saves the results to a file which is pulled by the software.

# cat /root/speedtest.txt
Ping: 29.792 ms
Download: 113.56 Mbit/s
Upload: 17.67 Mbit/s

Have you been able to clean that up to only get the numbers?

**Comment by dirdled** [comment]
The output that I save is just exactly like that, but when RPIMonitor reads the file it runs it though this line

<code>
cat $speedTestOut|sed -n "/$dataType/Is/.*: \(.*\) .*/\1/p"
</code>
<br>Where <br>$speedTestOut is the name of the output file (it looks like /root/speedtest.txt in your case)
<br>and <br>$datatype is either ping, download, or upload

**Comment by sgrayban** [comment]
@dirdled thanks !!!

![image](https://user-images.githubusercontent.com/1351910/147301012-9086a516-1bca-42a5-b177-da62c512c122.png)

My script has this now....
```
#!/bin/bash
# file: /root/speedtest.sh

speedtest --simple > /root/speedtest.txt
cat /root/speedtest.txt|sed -n "/ping/Is/.*: \(.*\) .*/\1/p" > /root/ping.txt
cat /root/speedtest.txt|sed -n "/download/Is/.*: \(.*\) .*/\1/p" > /root/download.txt
cat /root/speedtest.txt|sed -n "/upload/Is/.*: \(.*\) .*/\1/p" > /root/upload.txt

```
Now to get that into the graph stats!!

**Comment by dirdled** [comment]
@sgrayban  Nice... If I can figure out how to to a private message on here, I can share the whole script and also show you what I have for a configuration to use the script...But I'm not sure my fragile ego can handle having people who really know what they're doing critique my code (I guess I'm assuming that you're OK to show because you're asking me questions :) )

**Comment by sgrayban** [comment]
@dirdled no PM's here. Are you on Facebook?

**Comment by sgrayban** [comment]
@dirdled and just to tease you more

![image](https://user-images.githubusercontent.com/1351910/147311044-ddaf7f3e-2552-4e36-8bcd-fba436f02c99.png)

Got the icons working

**Comment by sgrayban** [comment]
@dirdled also I forked this repo and started to add my changes --> https://github.com/sgrayban/RPi-Monitor

**Comment by dirdled** [comment]
@sgrayban Yeah, I'm on facebook (same ID), I don't login often, but I have an account

Here's what mine looks like

![image](https://user-images.githubusercontent.com/6250243/147316004-85511733-b94d-4381-8d28-57918f235b80.png)

Looks like they bumped my upload speed, so I should adjust my gauge settings

**Comment by sgrayban** [comment]
My FB account https://www.facebook.com/Shaul.ben.Yeshua.1 -- PM me there

**Comment by Ajimaru** [comment]
No worries, poste your script, we are curious.

**Comment by sgrayban** [comment]
@Ajimaru @dirdled did send it to me and it's a nifty bit of code. I don't know why he doesn't wasn't to share it publicly. Maybe he will reconsider with my praise??

**Comment by dirdled** [comment]
Ok, Ok, Since it sounds like @sgrayban doesn't think the code is a dumpster fire I'll share... Here are 2 files one is the speedtest Wrapper (in bash), and the other is the config file for Rpi-Monitor.
[SpeedChk.txt](https://github.com/XavierBerger/RPi-Monitor/files/7774845/SpeedChk.txt)
[internet.conf.txt](https://github.com/XavierBerger/RPi-Monitor/files/7774846/internet.conf.txt)

**Comment by sgrayban** [comment]
![image](https://user-images.githubusercontent.com/1351910/147366035-742b4c42-5322-4693-afff-89a8e70c0c7b.png)

**Comment by xgenosis** [comment]
> Ok, Ok, Since it sounds like @sgrayban doesn't think the code is a dumpster fire I'll share... Here are 2 files one is the speedtest Wrapper (in bash), and the other is the config file for Rpi-Monitor. [SpeedChk.txt](https://github.com/XavierBerger/RPi-Monitor/files/7774845/SpeedChk.txt) [internet.conf.txt](https://github.com/XavierBerger/RPi-Monitor/files/7774846/internet.conf.txt)

So this is pretty old but I'm trrying to get this up and running and i cannot.  I have setup both files but does the speedchk need to run on a timer?
